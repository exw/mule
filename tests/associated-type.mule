{ x = {} } : { type t = {}, x : t }
