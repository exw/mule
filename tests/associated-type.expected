
Desugared:

	(App
	 ((WithType
	   (Record (r_info ())
	    (r_types
	     (()
	      ((t
	        (Recur
	         (() t
	          (Record (r_info ()) (r_types (() () ())) (r_values (() () ())))))))
	      ()))
	    (r_values (() ((x (Var (() t)))) ()))))
	  (App
	   ((Fix Record)
	    (Lam ($0 (App ((App ((Update (Value x)) EmptyRecord)) EmptyRecord))))))))

inferred type:

	(Record (r_info ())
	 (r_types
	  (() ((t (Record (r_info ()) (r_types (() () ())) (r_values (() () ())))))
	   ()))
	 (r_values
	  (() ((x (Record (r_info ()) (r_types (() () ())) (r_values (() () ())))))
	   ())))

Runtime term:

	(App ((Fix Record) (Lam (0 () (Record ((x (Record ()))))))))

Evaluated:

	(Record ((x (Record ()))))
